File: labs/lab2.2/csharp/Program.cs
namespace Lab2_2;

static class Program {
    static void Main() {
        MyString str1 = new("Hello world!");
        MyString str2 = "What's poppin";
        string castedToString = (string)str1;
        Console.WriteLine($"String 1: {str1.Value}\nString 2: {str2.Value}\nString 1 casted to string type: {castedToString}");

        Numbers num1 = new("12345");
        Numbers num2 = new(420);
        int castedToInt = (int)num1;
        num1.RemoveSymbol(5);
        Console.WriteLine($"Number 1: {num1.Value}\nNumber 2: {num2.Value}\nNumber 1 casted to int type: {castedToInt}");
        Console.WriteLine("Looping over num1");
        foreach (char i in num1) Console.WriteLine(i);

        Lowercased lower1 = new("lowercase");
        Lowercased lower2 = Lowercased.ToLowercase("FROM UPPPERCASED");
        lower1.RemoveSymbol('a');
        Console.WriteLine($"Lower 1: {lower1.Value}\nLower 2: {lower2.Value}");
    }
}

File: labs/lab2.2/cpp/Program.cpp
#include "MyString.cpp"
#include "iostream"

using namespace Lab2_2;
using std::cout;

int main() {
    MyString str1("Hello world");
    MyString str2 = "What's poppin";
    string castedToString = (string)str2;
    cout << "String 1:" << str1.GetValue()
        << " String 2:" << str2.GetValue()
        << " MyString casted to string type: " << castedToString
        << '\n';
    
    Numbers num1("12345");
    Numbers num2(420);
    int castedToInt = (int)num1;
    num1.RemoveSymbol('5');
    cout << "Number 1:" << num1.GetValue()
        << " Number 2:" << num2.GetValue()
        << " Numbers casted to int type: " << castedToInt
        << '\n';

    
    Lowercased lower1("hallo");
    Lowercased lower2 = Lowercased::ToLowercased("YOLO FELLOW KIDS");
    lower1.RemoveSymbol('a');
    cout << "Lowercase string 1:" << lower1.GetValue()
        << " Lowercase string 2:" << lower2.GetValue()
        << '\n';
    return 0;
}